src/utils.ts

```diff
@@ -266,3 +266,99 @@ export const proxyWithComputed = <T extends object, U extends object>(
   const proxyObject = proxy(initialObject) as T & U
   return proxyObject
 }
+
+export type WatchGet = <T extends object>(value: T) => T
+export type WatchCallback = (get: WatchGet) => (() => void) | void | undefined
```

Comment:
```suggestion
type WatchGet = <T extends object>(value: T) => T
type WatchCallback = (get: WatchGet) => (() => void) | void | undefined
```
I prefer hiding types if that's not too much trouble, which allows us to refactor things without hustle.

Reply:
Sure thing, will fix.